File: ruby/hamming/hamming.rb
class Hamming
  VERSION = 1
  
  def Hamming.compute(dna_1 = "AG", dna_2 = "CT")
    dna1 = dna_1.length
    dna2 = dna_2.length
    
    raise ArgumentError, "String must be of equal length." if dna1 != dna2
    
    count = 0
    for i in 0..dna1
      if dna_1[i] != dna_2[i]
        count += 1
      end
    end
    return count    
  end  
end

Hamming.compute

File: javascript/hamming/hamming.js
var compute = function() {};

compute = function(string1, string2) {

    if (string1.length !== string2.length)
	throw 'DNA strands must be of equal length.';
    
    var hamming = 0;
    for (var i = 0; i < string1.length; i++){
	if (string1[i] !== string2[i])
	    hamming++;
    }

    return hamming;
};

module.exports = compute;
